pyat/at/lattice/lattice_object.py

```diff
@@ -529,6 +529,31 @@ def copy(self) -> "Lattice":
     def deepcopy(self) -> "Lattice":
         """Returns a deep copy of the lattice"""
         return copy.deepcopy(self)
+        
+    def slice_elements(self, refpts, slices: Optional[int] = 1) \
+            -> "Lattice":
+        """Create a new lattice by slicing the elements at refpts
+
+        Keyword arguments:
+            size=None:      Length of a slice. Default: computed from the
+              range and number of points: ``size = (s_max-s_min)/slices``.
+            slices=1:       Number of slices in the specified range. Ignored if
+              size is specified. Default: no slicing
+
+        Returns:
+            newring:    New Lattice object
+        """
+        def slice_generator(_):
+            check = get_bool_index(self, refpts)
+            for el, ok in zip(self, check):
+                if ok:
```

Comment:
By analogy with the `slice` method, I would check:
```python
if ok and (slice > 1):
```
since 1 is the default value. Though I can't remember why one would use slice == 1…